src/main/java/io/debezium/connector/vitess/pipeline/txmetadata/VitessEpochProvider.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.vitess.pipeline.txmetadata;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.debezium.connector.vitess.Vgtid;
+
+public class VitessEpochProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(VitessEpochProvider.class);
+    private Map<String, Long> shardToEpoch = new HashMap<>();
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    public Long getEpoch(Long previousEpoch, String previousTransactionId, String transactionId) {
```

Comment:
If id represents one shard, maybe the variable name can just be called previousGtid to avoid confusion.

Also this method seems can be a `static` method, it doesn't use any instance variables.

Reply:
> Is the previousTransactionId and transactionId here represents one shard or multiple shards?
Updated variable name, it represents single shard gtid

> Also this method seems can be a static method, it doesn't use any instance variables.

It uses the shardToEpoch instance variable, so it cannot be static. However, the rank provider method could be static so I made that one static.